From: Ping Cheng <pinglinux@gmail.com>
Date: Thu, 28 Jun 2012 16:47:30 -0700
Subject: Input: wacom - don't retrieve touch_max when it is predefined

commit 1cecc5cc0658e128bcad0b29edb96f286066571d upstream.

Some models, such as 0xE6, report more fingers than we process.

Reported-by: Jonathan Nieder <jrnieder@gmail.com>
Signed-off-by: Ping Cheng <pingc@wacom.com>
Tested-by: Nils Kanning <nils@kanning.de>
Tested-by: Rafi Rubin <rafi@seas.upenn.edu>
Reviewed-by: Jason Gerecke <killertofu@gmail.com>
Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>
---
 drivers/input/tablet/wacom_sys.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: linux-3.2.46/drivers/input/tablet/wacom_sys.c
===================================================================
--- linux-3.2.46.orig/drivers/input/tablet/wacom_sys.c	2013-07-26 19:20:41.000000000 +0000
+++ linux-3.2.46/drivers/input/tablet/wacom_sys.c	2013-07-26 19:20:42.000000000 +0000
@@ -398,7 +398,9 @@
 				break;
 
 			case HID_USAGE_CONTACTMAX:
-				wacom_retrieve_report_data(intf, features);
+				/* leave touch_max as is if predefined */
+				if (!features->touch_max)
+					wacom_retrieve_report_data(intf, features);
 				i++;
 				break;
 			}
